### PR TITLE
Create custom stripped CZ gate

### PR DESCRIPTION
**Pull request by SalahedeenIssa**
Fixes https://github.com/SupertechLabs/cirq-superstaq/issues/412

### CODE REVIEW

**Comment by perlinm** [comment]
fixes #412

File: cirq_superstaq/custom_gates.py [hunk 1]
```diff
@@ -572,6 +572,28 @@ def _json_dict_(self) -> Dict[str, Any]:
...
+
+    def _num_qubits_(self) -> int:
+        return 2
+
+    def _unitary_(self) -> Optional[npt.NDArray[np.complex_]]:
+        return np.array(
+            [
+                [1.0, 0.0, 0.0, 0.0],
+                [0.0, 1.0, 0.0, 0.0],
+                [0.0, 0.0, 1.0, 0.0],
```
Comment (perlinm) [inline cirq_superstaq/custom_gates.py:588]: I think these should be phases too
```suggestion
                [0.0, np.exp(1j * self.phi), 0.0, 0.0],
                [0.0, 0.0, np.exp(1j * self.phi), 0.0],
```

File: cirq_superstaq/custom_gates.py [hunk 2]
```diff
@@ -572,6 +572,28 @@ def _json_dict_(self) -> Dict[str, Any]:
         return cirq.protocols.obj_to_dict_helper(self, ["theta", "phi"])
 
 
+class StrippedCZGate(cirq.Gate):
+    def __init__(self, phi=0) -> None:
+        super(StrippedCZGate, self)
```
Comment (perlinm) [inline cirq_superstaq/custom_gates.py:577]: What does this line do?

Comment (SalahedeenIssa) [reply cirq_superstaq/custom_gates.py:577]: I wanted to stay as close to the format shown in this [https://quantumai.google/cirq/build/custom_gates](link) as possible to start off, and then make changes as necessary.

File: cirq_superstaq/custom_gates.py (hunk 1 above)
Comment (SalahedeenIssa) [reply cirq_superstaq/custom_gates.py:588]: Ah right good catch.

**Comment by SalahedeenIssa** [comment]
New branch made instead

**Comment by perlinm** [comment]
note: new branch is #441

**Pull request closed without merging**